### PR TITLE
Instructions to disable crc-check function

### LINKED ISSUE #100: Update installation procedure and troubleshooting
In regard to https://github.com/ros-industrial/motoman/discussions/562#discussioncomment-6482427
We need to update the order of operations during setup and add a step to the troubleshooting. 

This is the same as https://github.com/ros-industrial/motoman/issues/563 for MotoROS1.

### PR DESCRIPTION
**Pull request by ted-miller**
Fixes #100 
Context: https://github.com/ros-industrial/motoman/discussions/562#discussioncomment-6482427

Disable CRC check before the Speed FB things attempts to modify parameters.

### CODE REVIEW

**Comment by ted-miller** [comment]
Hold up. Needs troubleshooting

File: README.md [hunk 1]
```diff
@@ -304,6 +314,13 @@ Reboot again and there should be no alarms.
 
 If you receive any errors or alarms after rebooting, please refer to the [Troubleshooting](#troubleshooting) section for information on how to remedy the issue.
 
+If `SAVE DATA CRC CHECK FUNC (FSU)` was disabled at the start of this procedure, then it can now be reenabled.
+
+ 1. upgrade to *SAFETY* security level by touching `[System Info]`→`[Security]` (default password is all `5`'s)
+ 1. touch `[Setup]`→`[Function Enable]`
+ 1. navigate to `SAVE DATA CRC CHECK FUNC (FSU)`
+ 1. set this feature to `VALID`
```
Comment (gavanderhoorn) [inline README.md:322]: should we perhaps put this above the "If you receive any errors or alarms .." sentence above it?

It's still part of the normal setup procedure, right? Then after this section, we explain where to find troubleshooting information.

Comment (ted-miller) [reply README.md:322]: I like where it is. If they get any errors, then it's likely that the Speed FB didn't execute. In that case, they need to leave the CRC check off while they troubleshoot the other problems.

**Comment by ted-miller** [comment]
I think it's good now.

**Comment by gavanderhoorn** [comment]
Linter isn't happy.

**Comment by ted-miller** [comment]
Oh... Do I need to fix the commit itself instead of the just the file?

**Comment by gavanderhoorn** [comment]
Well, you could just force-push a fixed version, or add a commit that fixes it.

File: README.md [hunk 2]
```diff
@@ -246,6 +246,16 @@ Insert the storage device into the robot's programming pendant and refer to the
 
 ### DX200, YRC1000, and YRC1000micro
 
+If the controller is configured with the Functional Safety Unit (FSU), then `SAVE DATA CRC CHECK FUNC (FSU)` must be temporarily disabled during the installation procedure.
+
+In *Normal Operation* mode:
```
Comment (gavanderhoorn) [inline README.md:251]: Maybe we've already explained this somewhere, but how do you start the controller in "normal operation" mode?

File: doc/troubleshooting.md [hunk 3]
```diff
@@ -153,13 +153,22 @@ This alarm occurs if the FSU is enabled when installing MotoROS2.
 The MotoROS2 driver attempts to enable the *Speed Feedback* parameters, but is unable to change the required parameters due to FSU settings.
 You must temporarily disable the CRC check for the *Speed Feedback* update to complete.
 
-Touch `[RESET]` to clear the active alarm.
+Boot the controller into *Maintenance* mode by holding `{Main Menu}` on the keypad.
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:156]: After we've turned it off, right? Or does just holding `{Main Menu}` reboot the controller for me ? :)

File: doc/troubleshooting.md [hunk 4]
```diff
@@ -153,13 +153,22 @@ This alarm occurs if the FSU is enabled when installing MotoROS2.
...
+
+Boot the controller into *Normal Operation* mode
 Switch the pendant to TEACH mode.
 Then, from the main menu, touch `[System]`→`[Security]` and upgrade to *SAFETY* security level (the default password is all `5`s).
 Then touch `[Setup]`→`[Function Enable]`.
 Navigate to *SAVE DATA CRC CHECK FUNC (FSU)*.
-Set this feature to *INVALID* and reboot the controller in normal mode.
+Set this feature to *INVALID*.
+
+Boot the controller into *Maintenance* mode by holding `{Main Menu}` on the keypad.
```
Comment (gavanderhoorn) [inline doc/troubleshooting.md:168]: Should we add a "Turn off the controller" step here?

I can see people getting confused, as "nothing happens", but they are "holding the `{Main Menu}` button after "setting the feature to *INVALID*".

**Comment by gavanderhoorn** [comment]
@ted-miller: the solution text for `4997` became a bit long and hard to follow.

I've restructured it a bit. Could you check [the rendered version](https://github.com/Yaskawa-Global/motoros2/blob/crc_check/doc/troubleshooting.md#alarm-49974) and see whether you agree and it's all correct?

File: doc/troubleshooting.md [hunk 5]
```diff
@@ -153,28 +153,41 @@ This alarm occurs if the FSU is enabled when installing MotoROS2.
...
+- Boot the controller into *Maintenance* mode by powering on the controller while holding `{Main Menu}` on the keypad.
+- Reinstall the MotoROS2 `.out` file following the [regular installation procedure](https://github.com/Yaskawa-Global/motoros2#installation).
+- Reboot the robot controller and let it boot into *Normal Operation* mode (do not hold any keys).
+- After rebooting, you should see the alarm: `Alarm 8001 [10]` ("Speed FB enabled, reboot now").
+- *Speed Feedback* has now permanently been enabled.
 
-Reboot once more and follow the steps above to re-enable the *SAVE DATA CRC CHECK FUNC (FSU)*.
+Re-enable the CRC check:
+
+- Reboot once more and follow the steps in the *Disable the CRC check* section above to re-enable the *SAVE DATA CRC CHECK FUNC (FSU)* (be sure to set it to *VALID*).
```
Comment (ted-miller) [inline doc/troubleshooting.md:184]: You don't need to reboot here.

Comment (ted-miller) [reply doc/troubleshooting.md:184]: I see that my text said that too. It was wrong though

File: doc/troubleshooting.md [hunk 6]
```diff
@@ -153,28 +153,41 @@ This alarm occurs if the FSU is enabled when installing MotoROS2.
...
+
+No more alarms should be raised and MotoROS2 should now be installed.
+
+If the alarm is raised again, save a copy of the `PANELBOX.LOG` from the robot's teach pendant.
```
Comment (ted-miller) [inline doc/troubleshooting.md:188]: I'm not sure what we would do with a panelbox. Maybe better to get an ALL.PRM.

File: doc/troubleshooting.md (hunk 5 above)
Comment (ted-miller) [reply doc/troubleshooting.md:184]: nvm. I'm thinking of the wrong step. Ignore this

**Review by gavanderhoorn — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
This seems :+1: to me now, but seeing as I added commits myself, I'd like a review from you as well @ted-miller.

**Comment by gavanderhoorn** [comment]
I'll squash merge this after your review @ted-miller. No need to merge all the fixup commits.

File: doc/troubleshooting.md [hunk 7]
```diff
@@ -185,8 +185,8 @@ Re-enable the CRC check:
 
 No more alarms should be raised and MotoROS2 should now be installed.
 
-If the alarm is raised again, save a copy of the `PANELBOX.LOG` from the robot's teach pendant.
-Open a new issue on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues), describe the problem and attach the `PANELBOX.LOG` to the issue.
+If the alarm is raised again, save a copy of the `ALL.PRM` from the robot's teach pendant.
+Open a new issue on the [Issue tracker](https://github.com/yaskawa-global/motoros2/issues), describe the problem and attach the `ALL.PRM` to the issue (alternatively: send it to the MotoROS2 developers in a private email).
```
Comment (ted-miller) [inline doc/troubleshooting.md:189]: > (alternatively: send it to the MotoROS2 developers in a private email).

I'm curious what the reason for this is? There isn't any proprietary in a parameter file.

Comment (gavanderhoorn) [reply doc/troubleshooting.md:189]: I just wanted to pre-empt potential questions about this. I've had many such 'private emails' sent to me just because a robot was being used in a 'sensitive context' (or commercial) and *everything* was considered either proprietary or confidential.

**Comment by ted-miller** [comment]
It won't let me push the 'approve' button. But it looks good. Squash it.
